web/api/webrpc/pdp.go

```diff
@@ -245,14 +245,63 @@ type FSRegistryStatus struct {
 }
 
 type FSPDPOffering struct {
-	ServiceURL                 string `json:"service_url"`
-	MinPieceSizeInBytes        int64  `json:"min_size"`
-	MaxPieceSizeInBytes        int64  `json:"max_size"`
-	IpniPiece                  bool   `json:"ipni_piece"`
-	IpniIpfs                   bool   `json:"ipni_ipfs"`
-	StoragePricePerTibPerMonth int64  `json:"price"`
-	MinProvingPeriodInEpochs   int64  `json:"min_proving_period"`
-	Location                   string `json:"location"`
+	ServiceURL               string `json:"service_url"`
+	MinPieceSizeInBytes      int64  `json:"min_size"`
+	MaxPieceSizeInBytes      int64  `json:"max_size"`
+	IpniPiece                bool   `json:"ipni_piece"`
+	IpniIpfs                 bool   `json:"ipni_ipfs"`
+	StoragePricePerTibPerDay int64  `json:"price"`
+	MinProvingPeriodInEpochs int64  `json:"min_proving_period"`
+	Location                 string `json:"location"`
+	PaymentTokenAddress      string `json:"payment_token_address"`
+}
+
+// capabilitiesToOffering converts contract capabilities to FSPDPOffering and remaining custom capabilities
+func capabilitiesToOffering(keys []string, values [][]byte) (*FSPDPOffering, map[string]string) {
+	offering := &FSPDPOffering{}
+	customCaps := make(map[string]string)
+
+	for i, key := range keys {
+		if i >= len(values) {
+			break
+		}
+		value := values[i]
+
+		switch key {
+		case contract.CapServiceURL:
+			offering.ServiceURL = string(value)
+		case contract.CapMinPieceSize:
+			offering.MinPieceSizeInBytes = new(big.Int).SetBytes(value).Int64()
+		case contract.CapMaxPieceSize:
+			offering.MaxPieceSizeInBytes = new(big.Int).SetBytes(value).Int64()
+		case contract.CapIpniPiece:
+			offering.IpniPiece = len(value) > 0 && value[0] == 0x01
+		case contract.CapIpniIpfs:
+			offering.IpniIpfs = len(value) > 0 && value[0] == 0x01
```

Comment:
These keys are optional, so if they exist, can treat the capability as declared.
```suggestion
			offering.IpniPiece = true
		case contract.CapIpniIpfs:
			offering.IpniIpfs = true
```